File: web-client-examples/src/main/ruby/io/vertx/example/webclient/oauth/twitter_o_auth_example.rb
require 'vertx-web-client/web_client'
require 'vertx-web-common/body_codec'
@AUTH_URL = "https://api.twitter.com/oauth2/token"
@TWEET_SEARCH_URL = "https://api.twitter.com/1.1/search/tweets.json"
@B64_ENCODED_AUTH = "base64(your-consumer-key:your-consumer-secret)"

# Create the web client.
client = VertxWebClient::WebClient.create($vertx)

queryToSearch = "vertx"

# First we need to authenticate our call.
authHeader = "Basic #{@B64_ENCODED_AUTH}"
client.post_abs(@AUTH_URL).as(VertxWebCommon::BodyCodec.json_object()).add_query_param("grant_type", "client_credentials").put_header("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8").put_header("Authorization", authHeader).send() { |authHandler_err,authHandler|
  # Authentication successful.
  if (authHandler_err == nil && 200 == authHandler.status_code())
    authJson = authHandler.body()
    accessToken = authJson['access_token']
    header = "Bearer #{accessToken}"
    # Making call to search tweets.
    client.get_abs(@TWEET_SEARCH_URL).as(VertxWebCommon::BodyCodec.json_object()).add_query_param("q", queryToSearch).put_header("Authorization", header).send() { |handler_err,handler|
      if (handler_err == nil && 200 == handler.status_code())
        puts handler.body()
      else
        puts handler_err.get_message()
      end
    }
  else
    puts authHandler_err.get_message()
  end
}

File: web-client-examples/src/main/js/io/vertx/example/webclient/oauth/twitter_o_auth_example.js
var WebClient = require("vertx-web-client-js/web_client");
var BodyCodec = require("vertx-web-common-js/body_codec");
var AUTH_URL = "https://api.twitter.com/oauth2/token";
var TWEET_SEARCH_URL = "https://api.twitter.com/1.1/search/tweets.json";
var B64_ENCODED_AUTH = "base64(your-consumer-key:your-consumer-secret)";

// Create the web client.
var client = WebClient.create(vertx);

var queryToSearch = "vertx";

// First we need to authenticate our call.
var authHeader = "Basic " + B64_ENCODED_AUTH;
client.postAbs(AUTH_URL).as(BodyCodec.jsonObject()).addQueryParam("grant_type", "client_credentials").putHeader("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8").putHeader("Authorization", authHeader).send(function (authHandler, authHandler_err) {
  // Authentication successful.
  if (authHandler_err == null && 200 === authHandler.statusCode()) {
    var authJson = authHandler.body();
    var accessToken = authJson.access_token;
    var header = "Bearer " + accessToken;
    // Making call to search tweets.
    client.getAbs(TWEET_SEARCH_URL).as(BodyCodec.jsonObject()).addQueryParam("q", queryToSearch).putHeader("Authorization", header).send(function (handler, handler_err) {
      if (handler_err == null && 200 === handler.statusCode()) {
        console.log(handler.body());
      } else {
        console.log(handler_err.getMessage());
      }
    });
  } else {
    console.log(authHandler_err.getMessage());
  }
});
